src/components/SingleArticle/SingleArticle.jsx

```diff
@@ -1,31 +1,53 @@
-import { getArticleById } from "../../utils/api";
+import { getArticleById, getCommentsById } from "../../utils/api";
 import React, { useState, useEffect } from "react";
 import { useParams } from "react-router-dom";
 import "./SingleArticle.css";
+
 function SingleArticle() {
   const { article_id } = useParams();
   const [article, setArticle] = useState(null);
+  const [comments, setComments] = useState([]);
 
   useEffect(() => {
     getArticleById(article_id).then((data) => {
       setArticle(data.article);
     });
+
+    getCommentsById(article_id).then((data) => {
+      setComments(data.comments);
+    });
   }, [article_id]);
```

Comment:
You don't need the article id in the dependancy array as it cannot change on it's own. It would be changed by navigating to another page. You need the array for sure so you don't get a loop but nothing needs to be in it.